js/piwik.js

```diff
@@ -4193,42 +4193,21 @@ if (typeof window.Piwik !== 'object') {
                     request += '&ec_dt=' + discount;
                 }
 
-                if (ecommerceItems) {
-                    // Removing the SKU index in the array before JSON encoding
-                    for (sku in ecommerceItems) {
-                        if (Object.prototype.hasOwnProperty.call(ecommerceItems, sku)) {
-                            // Ensure name and category default to healthy value
-                            if (!isDefined(ecommerceItems[sku][1])) {
-                                ecommerceItems[sku][1] = "";
+                if (ecommerceItems.length) {
+                    ecommerceItems.forEach(function(ecommerceItem) {
```

Comment:
.forEach might not be supported in all browsers, the tracker is supposed to work even in older browsers. Can you maybe rewrite it to use a plain for loop?